content/en/blog/general/psp-migration/index.md

```diff
@@ -0,0 +1,366 @@
+
+**Limited to User, Group, and ServiceAccount.** As it was tied to RBAC, PSP had its scope of application tied to these principals which made it difficult to apply and often resulted in a huge web of these mappings. PSA is limited to either the whole cluster or Namespace, but nothing further. Kyverno can be associated with anything and apply as granular as you can express in Kubernetes.
+
+**No support for Pod controllers.** Another bummer for PSP users was the fact that Pod controllers such as Deployments and StatefulSets would always be allowed but Pods would be blocked. You had to figure this out typically by going to inspect the ReplicaSet to understand why. PSA has basically the same behavior here. Kyverno has automatic support for Pod controllers through its ability to [auto-generate Pod controller rules](/docs/writing-policies/autogen/).
+
+**No auditing.** A PSP either blocks or it doesn't, there's no in between. PSA can audit, warn, or block, but the audits are locked up in the API server's audit log which isn't visible inside the cluster. Kyverno generates its audits as a cluster-visible, custom resource called a [Policy Report](/docs/policy-reports/) which is an open standard developed by the Kubernetes Policy Working Group and is adopted by other tools aside from just Kyverno. These Policy Reports are just another Kubernetes resource allowing them to be scraped by other tools and, importantly, read by other users in a way which can be controlled via RBAC.
+
+**No message customization.** When a PSP blocks a Pod the message can't be customized. Same situation with PSA. The message returned by Kyverno can be fully customized making it very useful to show, for example, a team name or email of someone for developers or users to contact.
+
+**Integrated.** PSP and PSA are both integrated into the control plane. Kyverno is an external component (a dynamic admission controller) and so must be run separately in your cluster. In order to do this, Kyverno uses webhooks.
+
+## Migration Outline
+
+When migrating from PSP to Kyverno, this is the high-level approach that we recommend. Links are provided for the Kyverno concepts which are applicable to this migration, however it is still recommended that before diving into a migration yourself you spend some more time with the documentation and, of course, experimentation in a lab or non-production environment.
```

Comment:
Should we recommend scanning the cluster with the CLI as a first step?

```sh
kustomize build https://github.com/kyverno/policies/pod-security/enforce | kubectl kyverno apply --cluster  -
```

Reply:
It's a good idea. But the output produced from that exact command, even on a small cluster, is going to be overwhelming. Even on a small K3d cluster, the output had over 65 failed results.

How about if we recommended scanning with just the baseline profile using the podSecurity subrule instead? It'll produce a far shorter list and probably a more important one for readers to focus their attention.

```sh
curl -s https://raw.githubusercontent.com/kyverno/policies/main/pod-security/subrule/podsecurity-subrule-baseline/podsecurity-subrule-base
line.yaml | kubectl kyverno apply --cluster -
```

WDYT?

Reply:
Yes, scanning with `baseline` first and then progressing to `restricted` makes sense as a recommended approach.

The CLI can also scan one namespace as a time. 

Its missing a nicer report format (SARIF, HTML, etc.), which would make it a lot more usable.

Reply:
Thanks, just pushed this update.